drivers/hmis/app/graphql/mutations/submit_form.rb

```diff
@@ -24,20 +24,24 @@ def resolve(input:)
       # Find or create record
       if input.record_id.present?
         record = klass.viewable_by(current_user).find_by(id: input.record_id)
+        entity_for_permissions = record # If we're editing an existing record, always use that as the permission base
       else
-        record = klass.new(
-          **related_id_attributes(klass.name, input),
-          data_source_id: current_user.hmis_data_source_id, # Not all records actually have this, but we need it to check permissions for some of them
-        )
+        entity_for_permissions, attributes = permission_base_and_attributes(klass.name, input, current_user.hmis_data_source_id)
```

Comment:
This is super optional but I wonder if we should take the opportunity to fix up the old logic in permission_base_and_record a little. Some things we could adjust:
* Why pass in klass.name and not klass? Using the klass itself seems more robust against the class names changing under us since you'd get an error instead or a silent failure
* Maybe this method should return the new record rather than attrs? 🤷 

You might end up with something like this:
```ruby
  def resolve
    entity_for_permissions, record = permission_base_and_record(klass, input, current_user.hmis_data_source_id)
    ...
  end

  private def permission_base_and_attributes(klass, input, data_source_id)
    # ...
    case klass
    when Hmis::Hud::Client
        [nil, klass.new(ds)]
        ....etc
    end
  end

```

Reply:
Thanks! I updated it to return the record, that feels cleaner. I'm not sure about using the strings vs class name, it's this way because that's how the class name is specified on the Definition. I left that one as-is just to minimize changes